packages/ui-components/src/components/layout/tabs/Horizontal.vue

```diff
@@ -0,0 +1,238 @@
+const buttonContainer = ref(null as Nullable<HTMLDivElement>)
+const scrollContainer = ref<HTMLElement | null>(null)
+const showLeftArrow = ref(false)
+const showRightArrow = ref(false)
+const isInitialSetup = ref(true)
+
+const buttonClass = computed(() => {
+  return (item: LayoutPageTabItem) => {
+    const isActive = activeItem.value?.id === item.id
+    const baseClasses = [
+      'relative',
+      'z-10',
+      'flex',
+      'items-center',
+      'disabled:opacity-60 disabled:hover:border-transparent disabled:cursor-not-allowed disabled:hover:bg-transparent',
+      'text-base',
+      'gap-1.5',
+      'hover:sm:border-outline-2',
+      'pb-2',
+      'border-b-[2px]',
+      'border-transparent',
+      'max-w-max',
+      'last:mr-6',
+      'whitespace-nowrap'
+    ]
+
+    if (isActive) baseClasses.push('text-primary', 'hover:text-primary')
+    else baseClasses.push('text-foreground')
+
+    return baseClasses
+  }
+})
+
+const activeItemRef = computed(() => {
+  const id = activeItem.value?.id
+  if (!id) return null
+
+  const parent = buttonContainer.value
+  if (!parent) return null
+
+  const btns = [...parent.getElementsByClassName('tab-button')] as HTMLElement[]
+  return btns.find((b) => b.dataset['tabId'] === id) || null
+})
+
+const borderStyle = computed<CSSProperties>(() => {
+  const element = activeItemRef.value
+  return {
+    left: `${element?.offsetLeft || 0}px`,
+    width: `${element?.clientWidth || 0}px`
+  }
+})
+
+const setActiveItem = (item: LayoutPageTabItem) => {
+  activeItem.value = item
+  isInitialSetup.value = false
+}
+
+const isActiveItem = (item: LayoutPageTabItem) => {
+  return activeItem.value?.id === item.id
+}
+
+const checkArrowsVisibility = () => {
+  const container = scrollContainer.value
+  if (!container) return
+
+  const scrollWidth = container.scrollWidth
+  const clientWidth = container.clientWidth
+  const scrollLeft = container.scrollLeft
+  const buffer = 1
+
+  showLeftArrow.value = scrollLeft > buffer
+  showRightArrow.value = scrollLeft < scrollWidth - clientWidth - buffer
+}
+
+const scrollLeft = () => {
+  scrollContainer.value?.scrollBy({ left: -100, behavior: 'smooth' }) // Adjust the scroll amount as needed
+  checkArrowsVisibility()
+}
+
+const scrollRight = () => {
+  scrollContainer.value?.scrollBy({ left: 100, behavior: 'smooth' }) // Adjust the scroll amount as needed
+  checkArrowsVisibility()
+}
+
+const handleScroll = () => {
+  checkArrowsVisibility()
+}
+
+const ensureActiveItemVisible = () => {
+  const activeButton = activeItemRef.value
+  if (activeButton && scrollContainer.value) {
+    activeButton.scrollIntoView({
+      behavior: 'smooth',
+      block: 'nearest',
+      inline: 'center'
+    })
+  }
+}
+
+onMounted(() => {
+  if (isClient) {
+    if (props.items.length && !activeItem.value) {
+      setActiveItem(props.items[0])
+    }
+    checkArrowsVisibility()
+    nextTick(() => {
```

Comment:
nextTick/setTimeouts are sketchy and unreliable, i'd evaluate if there's a way to avoid this